cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -137,6 +137,7 @@ class HydratorPlusPlusTopPanelCtrl{
   }
   onSaveDraft() {
     this.HydratorPlusPlusConfigActions.saveAsDraft();
+    this.checkNameError();
```

Comment:
If placed before line 139, the error messages are not set at this point and hence it does not validate the name error.
Also draft does not get saved if called without pipeline name as on line 735 in config-store.js in **saveAsDraft** function.